webapp/components/interop.js

```diff
@@ -319,32 +390,38 @@ class WPTInterop extends QueryBuilder(
     return labels;
   }
 
-  computeTestRuns(metadata) {
-    return metadata && metadata.test_runs;
-  }
-
-  computeTestPaths(fileMetrics) {
-    const paths = fileMetrics && fileMetrics.map(m => m.dir) || [];
+  computeTestPaths(searchResults) {
+    const paths = searchResults && searchResults.results.map(r => r.test) || [];
     return new Set(paths);
   }
 
-  computeFileMetrics(allMetrics) {
-    let fileMetrics = [];
-    for (const metric of allMetrics.data) {
-      if (this.computePathIsATestFile(metric.dir)) {
-        fileMetrics.push(metric);
+  precomputedInteropLoaded(precomputedInterop) {
+    const searchResults = {
+      runs: this.testRuns,
+      results: [],
+    };
+    if (precomputedInterop) {
+      for (const metric of precomputedInterop.data) {
+        if (this.computePathIsATestFile(metric.dir)) {
+          searchResults.results.push({
+            test: metric.dir,
+            interop: metric.pass_rates,
+          });
+        }
+      }
+      const q = this.search;
+      if (q  && q.length) {
+        searchResults.results = searchResults.results.filter(t => t.test.toLowerCase().includes(q));
```

Comment:
Looks like client-side filtering. Isn't there an existing single place where this is handled, if we change the rules for how it's matched, or move it back to the server?

Reply:
Interop wasn't moved to a server-side search when Results was, so this is legacy code. Note that I haven't authored it in this PR, simply moved it.